reconcile/change_owners/self_service_roles.py

```diff
@@ -211,6 +209,18 @@ def change_type_contexts_for_self_service_roles(
     return change_type_contexts
 
 
+def _build_approver(role_member: RoleMember) -> Approver:
+    if not role_member.org_username:
+        raise ValueError(
+            f"Role member {role_member} has no org_username and can not be used as approver"
+        )
```

Comment:
i removed the double check for org_username. `_build_approver` can now yield `None` if the passed `RoleMember` is not a valid approver for whatever reason

Reply:
oh, then this may make it cleaner if my understanding of python's pattern match is same as other languages:

```python
def _build_approver(role_member: RoleMember) -> Optional[Approver]:
    match role_member:
        case RoleUser(org_username, tag_on_merge_requests):
            return Approver(org_username, tag_on_merge_requests)
        case RoleBot(org_username) if org_username:
            return Approver(org_username, False)
        case _:
            return None
```

Reply:
TIL python has case guards. adapted it.
i made the function "public" so it does not looks strange to write tests for it.

Reply:
To sad, destructive pattern match is not supported by default for all objects, code block above only works for objects defined `__match_args__` (dataclass add it by default), for other objects, it has to be explicit like this:

```python
    match role_member:
        case RoleUser(
            org_username=org_username,
            tag_on_merge_requests=tag_on_merge_requests,
        ):
            return Approver(org_username, tag_on_merge_requests)
        case RoleBot(org_username=org_username) if org_username:
            return Approver(org_username, False)
        case _:
            return None

```